fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricTagProvider.java

```diff
@@ -146,7 +146,7 @@ public ItemTagProvider(FabricDataGenerator dataGenerator) {
 		public void copy(Tag.Identified<Block> blockTag, Tag.Identified<Item> itemTag) {
 			Tag.Builder blockTagBuilder = Objects.requireNonNull(this.blockTagBuilderProvider, "Pass Block tag provider via constructor to use copy").apply(blockTag);
 			Tag.Builder itemTagBuilder = this.getTagBuilder(itemTag);
-			blockTagBuilder.streamEntries().forEach(itemTagBuilder::add);
+			blockTagBuilder.streamEntries().filter((entry) -> entry.getEntry().canAdd(this.registry::containsId, (id) -> true)).forEach(itemTagBuilder::add);
```

Comment:
Sounds a bit unexpected, why would copy() silently ignore some tags?

Reply:
Say you have a tag called `c:vines`, containing all the many types of vines in the game. You want both a block tag and an item tag for this.

You have `minecraft:vine`, `minecraft:weeping_vines`, and `minecraft:twisting_vines`, sure. Those all have both block form and item form.

But then there are `minecraft:weeping_vines_plant`, `minecraft:twisting_vines_plant`, `minecraft:cave_vines`, and `minecraft:cave_vines_plant`. Those are all *only* available in block form.

So, when you copy your block tag, containing all 7 vine blocks, to an item tag, you want it to only contain the 3 that actually exist as items. Copying the other 4 ids would result in a crash as those items don't exist.

Many vanilla block tags have item tag equivalents that exclude items that don't exist. Take `minecraft:banners`. The block tag contains all 16 colours of 'standing' banners - and wall banners. The item tag contains just the 16 'standing' banners because the wall banners don't have an item form. (or, rather, their item form is the same as the standing one - the same could be said of the above vines example, where the `_plant` blocks' item forms are the same as their respective 'standard' one (and `cave_vines`' item form is `minecraft:glow_berries`) - but I digress)

Reply:
Ah ok, makes sense. Could you maybe add to the javadoc that only existing tags are copied?

Reply:
Added a note `Any block ids that do not exist in the item registry will be filtered out automatically.` to the javadoc